src/module.c

```diff
@@ -560,17 +579,19 @@ int ReplySeriesRange(RedisModuleCtx *ctx, Series *series, api_timestamp_t start_
             RedisModule_ReplyWithDouble(ctx, sample.value);
             arraylen++;
         } while (SeriesIteratorGetNext(&iterator, &sample) == CR_OK &&
-                    (maxResults == -1 || arraylen < maxResults));
+                    (maxResults == -1 || arraylen < maxResults));            
     } else {
         bool firstSample = TRUE;
         do {
-            if (sample.timestamp >= last_agg_timestamp + time_delta) {
+            if ((iterator.reverse == NO_OPT && sample.timestamp >= last_agg_timestamp + time_delta) ||
+                (iterator.reverse == REVERSE && sample.timestamp <= last_agg_timestamp - time_delta)) {
```

Comment:
@ashtul im confusing is reverse is a flag or and options? if its a flag we should use: `if (iterator.reverse == TRUE) ....` if its a options member we should do `if (iterator.opts & REVERSE) ....`

Reply:
This will change due to the revert we have done

Reply:
It is a bool. I'm not even sure how it will behave with bitwise `&`.
The only place where we use a flag here is in the chunk itself. otherwise, it is a bool.